### PR TITLE
[New Feature] Optionally Add StatsD Metrics Reporting

### PR DESCRIPTION
**Pull request by andrew-ferguson-hs**
Added an entirely new feature to enable the delivery of metrics captured using the Github Autostatus Plugin to a StatsD endpoint as a potential alternative to InfluxDB.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusAction.java [hunk 1]
```diff
@@ -128,7 +130,12 @@ public void close() {
             }
         });
     }
-    
+
+    /**
+     * Sets flag indicating whether notifications are for a declarative pipeline
+     *
+     * @return if pipeline is declaritive or not
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusAction.java]:
```suggestion
     * @return if pipeline is decalarative or not
```

File: src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusJobListener.java [hunk 2]
```diff
@@ -148,4 +151,12 @@ private long getBlockedTime(Run<?, ?> build) {
 
         return action == null ? 0 : action.getTimeBlocked();
     }
+
+    private static void log(Level level, String format, Object... args) {
+        getLogger().log(level, String.format(format, args));
+    }
+
+    private static Logger getLogger() {
+        return Logger.getLogger(GithubNotificationConfig.class.getName());
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusJobListener.java]:
```suggestion
        return Logger.getLogger(BuildStatusJobListener.class.getName());
```

File: src/main/java/org/jenkinsci/plugins/githubautostatus/GithubBuildStatusGraphListener.java [hunk 3]
```diff
@@ -386,6 +415,12 @@ BuildStatusAction buildStatusActionFor(FlowExecution exec) {
         return buildStatusAction;
     }
 
+    /**
+     * Get the jenkins run objection of the specified executing workflow
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/GithubBuildStatusGraphListener.java]:
```suggestion
     * Get the jenkins run object of the specified executing workflow
```

File: src/main/java/org/jenkinsci/plugins/githubautostatus/StatsdClient.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 jxpearce.
```
Comment (jeffpearce) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/StatsdClient.java]: `Copyright 2019 Tom Hadlaw

**Comment by andrew-ferguson-hs** [comment]
@jeffpearce Updated PR with the requested changes!

**Comment by andrew-ferguson-hs** [comment]
@jeffpearce Any chance you'll have time to give this another look in the near future?

**Comment by jeffpearce** [comment]
Sorry @andrew-ferguson-hs. Looking now. Thanks for the reminder

**Comment by jeffpearce** [comment]
Thanks for the PR! I'll make a new release. I can try to update the wiki page to describe it, but if you have the bandwidth, I suspect you'd be able to do a better job that iI can :-)

**Comment by andrew-ferguson-hs** [comment]
@jeffpearce Thanks for the quick reply! More than happy to write the wiki documentation for you, are you ready for that now or should we wait til _after_ you've made the new release?

**Comment by jeffpearce** [comment]
@andrew-ferguson-hs just released - version 3.5

**Comment by andrew-ferguson-hs** [comment]
@jeffpearce Awesome, thanks so much! I'll update the wiki documentation tomorrow during my workday (PDT) and let you know when it's complete

**Comment by andrew-ferguson-hs** [comment]
@jeffpearce Documentation updated! If you have any changes or any additional work you need from us, please let either myself or Shane know! It's been a pleasure working with you on this 👍

**Comment by jeffpearce** [comment]
Thanks for the docs @andrew-ferguson-hs - looks great! And many thanks for the added feature.

**Pull request merged**